### PR TITLE
move inactive fn-matching to demangled names

### PR DESCRIPTION
**Pull request by ZuseZ4**

### CODE REVIEW

**Comment by ZuseZ4** [comment]
ss5print might be swift? It probably should be translated then later.

File: enzyme/Enzyme/ActivityAnalysis.cpp [hunk 1]
```diff
@@ -266,6 +200,34 @@ const std::set<std::string> KnownInactiveFunctions = {
     "logbl",
 };
 
+const char *DemangledKnownInactiveFunctionsStartingWith[] = {
+    "f90io",
```
Comment (wsmoses) [inline enzyme/Enzyme/ActivityAnalysis.cpp:204]: Can this be a separate category. In particular the fortran f90io as an example doesn't belong here.

File: enzyme/Enzyme/ActivityAnalysis.cpp [hunk 2]
```diff
@@ -1438,8 +1412,9 @@ bool ActivityAnalyzer::isConstantValue(TypeResults const &TR, Value *Val) {
               F->getName() == "__fd_sincos_1") {
             return false;
           }
-          for (auto FuncName : KnownInactiveFunctionsStartingWith) {
-            if (F->getName().startswith(FuncName)) {
+          auto dName = demangle(F->getName().str());
```
Comment (wsmoses) [inline enzyme/Enzyme/ActivityAnalysis.cpp:1415]: Please make sure this works on LLVM 7/8

Comment (ZuseZ4) [reply enzyme/Enzyme/ActivityAnalysis.cpp:1415]: LLVM-7 demangle support seems too basic: https://github.com/llvm/llvm-project/blob/00b161b8971bc6d3cb55f13502288b8fe0dbaa42/llvm/include/llvm/Demangle/Demangle.h

So we now have the demangled support for C++ for >= 9 and the unmangled support for f90io, ss5print and (if <=8) for most C++ IO.

File: enzyme/Enzyme/ActivityAnalysis.cpp (hunk 1 above)
Comment (ZuseZ4) [reply enzyme/Enzyme/ActivityAnalysis.cpp]: afaik only ss5print and f90io are non-c++ names? So I moved them back to check unconditionally for those unmangled.

File: enzyme/Enzyme/ActivityAnalysis.cpp (hunk 2 above)
Comment (ZuseZ4) [reply enzyme/Enzyme/ActivityAnalysis.cpp:1490]: diff looks like I missed one place

Comment (ZuseZ4) [reply enzyme/Enzyme/ActivityAnalysis.cpp:1490]: fixed

**Review by tgymnich — APPROVED** [review]

**Pull request merged**